lib/linkify.js

```diff
@@ -0,0 +1,55 @@
+// github-style linkification for linkify-it
+//
+// Pass in a markdown-it parser instance and this will return a version fixed
+// up for the kind of linkification we're after.
+//
+// As far as I can determine, the basic algorithm is to linkify anything
+// starting with 'www.' and ending with a) a whitespace character, or b) a dot
+// followed by a whitespace character (in other words, include internal dots,
+// but exclude a trailing dot). There are plenty of cases where this results in
+// invalid links, but that's how they do it, so that's how we'll do it.
+//
+module.exports = function (parser) {
+  if (parser) {
+    var linkify = parser.linkify
+
+    linkify.set({fuzzyLink: false}) // turn off auto-linking normal hostnames
+    linkify.add('//', null) // turn off protocol-relative links
+
+    // linkify everything hostnamey starting with 'www.'
+    parser.linkify.add('www.', {
+      validate: function (text, pos, self) {
+        var match = text.slice(pos).match(/^\S*/)[0]
+
+        // there's probably a nice regex way of excluding a trailing dot from
+        // these matches, but it's simple enough to just trim them off here
+        if (match.slice(-1) === '.') match = match.slice(0, -1)
+
+        // if we have a zero-length match, i.e., if the original string was
+        // literally 'www.' all by itself, we return -1 because linkify-it does
+        // a boolean test on the result of the validator function (so we need a
+        // non-zero result here to make it through that test) and adds the
+        // returned value to an index it's using to point to the end of the
+        // matched substring. In this case, that means we'll move backwards one
+        // character and only link 'www' rather than the full 'www.' that was
+        // originally matched
+        return match.length || -1
+      },
+      normalize: function (match) {
+        match.url = 'http://' + match.url
+      }
+    })
+
+    // linkify-it's default behavior is to create matches against the
+    // registered schema prefixes when the character immediately preceding a
+    // potential match is non-alphanumeric; however, we also need to skip
+    // linkification when said character is punctuation. So here we rebuild the
+    // internal regular expressions the LinkifyIt instance will use.
+    var re = linkify.re
```

Comment:
only comment is the `re` is really short and maybe it's more readable to say `regex` or something a bit more elaborate. don't really have strong feels tho. the comment is super helpful.